src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.NonSourceDocumentProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tags;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
+{
+    internal partial class SuggestedActionsSourceProvider
+    {
+        /// <summary>
+        /// <see cref="ISuggestedActionsSourceProvider"/> for non-source documents,
+        /// i.e. <see cref="AdditionalDocument"/> and <see cref="AnalyzerConfigDocument"/>.
+        /// </summary>
+        [Export(typeof(ISuggestedActionsSourceProvider))]
+        [Export(typeof(NonSourceDocumentProvider))]
```

Comment:
Rather than having two separate types, can we just have one and use the defered creation to cover the regular Roslyn one too? We could just set the option on any file being opened.